src/infrastructure/adapters/vagas.repository.adapter.ts

```diff
@@ -29,13 +29,18 @@ export class VagasRepositoryAdapter implements VagasRepository {
     await this.vagaRepository.delete(id)
   }
 
-  async search(query: string): Promise<Array<Vaga>> {
-    return await this.vagaRepository.find({
+  async search(query: string, tipoDeficiencia?: string): Promise<Array<Vaga>> {
+    return (await this.vagaRepository.find({
       where: [
         { titulo: ILike(`%${query}%`) },
         { cargo: ILike(`%${query}%`) },
         { descricao: ILike(`%${query}%`) },
       ]
-    });
+    })).filter(
```

Comment:
Esse filtro deve ser feito na query SQL e não em memória através do método **filter**

Ou seja, tem que modificar o where para quando o tipoDeficiencia for passado, adicionar mais esse parâmetro na query.

Provavelmente vai ser necessário usar o QueryBuilder do TypeOrm pra isso:

[Usando where](https://typeorm.io/select-query-builder#adding-where-expression)

Ex:
![2024-09-26_19-28](https://github.com/user-attachments/assets/e69d1419-46c4-4d87-bec5-6ffd70ec0e58)
![2024-09-26_19-33](https://github.com/user-attachments/assets/ae95712f-cae3-4696-ad9d-c8a404e556ea)

Provavelmente vai ficar algo como
```
createQueryBuilder('vagas')
  .where('vagas.titulo like :titulo', {titulo: `%${query}%`})
  .orWhere('vagas.descricao like :descricao', {descricao: `%${query}%`})
  .orWhere('vagas.cargo like :cargo', {cargo: `%${query}%`})
  .andWhere('tipoDeficiencia in (:tipoDeficiencia)', {tipoDeficiencia: arrayTipoDeficiencia})

```